postroj/winrunner.py

```diff
@@ -60,45 +182,58 @@ def powershell(self, command):
         command = f"powershell -Command {command}"
         return self.run(command)
 
-    def run(self, command, strip_armor=True, translate_newlines=True):
-        click.echo(f"Running command: {command}")
-        command = f"docker --context={self.BOX} run -it --rm openjdk:17-windowsservercore-1809 {command}"
-        outcome = run(command)
-        if strip_armor:
-            """
-            b'\x1b[2J\x1b[?25l\x1b[m\x1b[H\r\n\r\n...\r\n\x1b[H\x1b]0;C:\\Windows\\system32\\cmd.exe\x00\x07\x1b[?25h\x1b[?25lhello \r\n\x1b[?25h'
-            """
-            prefix = "\x1b[?25h\x1b[?25l"
-            suffix = "\x1b[?25h"
-            cutoff_left = outcome.find(prefix) + len(prefix)
-            cutoff_right = outcome.rfind(suffix)
-            outcome = outcome[cutoff_left:cutoff_right]
-        if translate_newlines:
-            outcome = outcome.replace("\r\n", "\n")
-        # print(outcome.encode())
-        return outcome
+    def run(self, command, interactive: bool = False, tty: bool = False):
+        logger.info(f"Running guest command: {command}")
+
+        option_interactive = ""
+        if interactive or tty:
+            option_interactive = "-it"
+
+        # TODO: Propagate ``--rm`` appropriately.
+        command = f"docker --context={self.wdm_machine} run {option_interactive} --rm {self.image_real} {command}"
```

Comment:
It would be sweet to introduce the `--volume` option here, in order to bind-mount a directory from the host system. That is important for being able to continue editing within your working tree on your workstation, as usual.

Reply:
Deferred to #8.